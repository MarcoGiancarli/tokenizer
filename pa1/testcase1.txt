/*
 * This is a block comment and it shouldn't be recognized.
 */

// STRING/FLOAT LITERALS
"hey man y u do dis"
"y u make so many string";
"y u try to escape sequence strings lik dis \"stuff\" -- now is dis 1 string or 2 strings with stuff in the middle? u find out."
'm', 'a', '\k', '\e', 'a', '\s', ;

// WORDS & RESERVED WORDS
hello there these are words. words words words _words are __words__ are &words are w or ds;
words

// OPERATORS
loltheseareoperators#include%%%=%+++=+,---=-?x:0;;!=~///=/,***=*,^&&&=&|||=|.->-->
()[]{}<<<<=<=< >>>>=>=> &*

// NUMBERS
045 00124 66 0x5a54 5e6 7e-7 5e+7 4.3e7 3.5e-7 3.7e+0



// LET'S BREAK THINGS NOW
'\an, 'dw, 'ic, '\'', '', '\' lol
!!!

"THIS WILL NEVER END!!!

