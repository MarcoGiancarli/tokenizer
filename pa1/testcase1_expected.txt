string literal '"hey man y u do dis"'
string literal '"y u make so many string"'
end-of-statement operator ";"
string literal '"y u try to escape sequence strings lik dis \"stuff\" -- now is dis 1 string or 2 strings with stuff in the middle? u find out."'
char literal "'m'"
comma ","
char literal "'a'"
comma ","
char literal "'\k'"
comma ","
char literal "'\e'"
comma ","
char literal "'a'"
comma ","
char literal "'\s'"
comma ","
end-of-statement operator ";"
word "hello"
word "there"
word "these"
word "are"
word "words"
struct member operator "."
word "words"
word "words"
word "words"
word "_words"
word "are"
word "__words__"
word "are"
bitwise-and operator/address operator "&"
word "words"
word "are"
word "w"
word "or"
word "ds"
end-of-statement operator ";"
word "words"
word "loltheseareoperators"
include operator "#"
word "include"
mod operator "%"
mod operator "%"
mod-equals operator "%="
mod operator "%"
increment operator "++"
plus-equals operator "+="
addition operator "+"
comma ","
decrement operator "--"
minus-equals operator "-="
subtraction operator "-"
ternary question mark operator "?"
word "x"
ternary colon operator ":"
zero integer "0"
end-of-statement operator ";"
end-of-statement operator ";"
not-equals operator "!="
bitwise-not operator "~"
opening parenthesis "("
closing parenthesis ")"
opening bracket "["
closing bracket "]"
opening brace "{"
closing brace "}"
left-shift operator "<<"
left-shift-equals operator "<<="
less-than-or-equal-to operator "<="
less-than operator "<"
right-shift operator ">>"
right-shift-equals operator ">>="
greater-than-or-equal-to operator ">="
greater-than operator ">"
bitwise-and operator/address operator "&"
multiplication operator/pointer operator "*"
octal integer "045"
octal integer "00124"
decimal integer "66"
hexadecimal integer "0x5a54"
float with exponent "5e6"
float with exponent "7e-7"
float with exponent "5e+7"
float with exponent "4.3e7"
float with exponent "3.5e-7"
float with exponent "3.7e+0"
invalid char literal "'\an"
comma ","
invalid char literal "'dw"
comma ","
invalid char literal "'ic"
comma ","
char literal "'\''"
comma ","
invalid char literal "'',"
invalid char literal "'\' "
word "lol"
logical-not operator "!"
logical-not operator "!"
logical-not operator "!"
unended string literal ""THIS WILL NEVER END!!!"
